### DISCUSSION TITLE
How to set selected programmatically

### DISCUSSION THREAD

**Discussion by monicai18n** [JedWatson/react-select#5072] [Q&A]
I will be getting some form data and would like to set the data in the react state. Once I have the value in the state how can I set selected 
programmatically in react-select on single select and multi select. 

I also tried something like this in render:
getOptionValue={this.getOptionValue}

function:
getOptionValue = option => option.key;

and setting value like this in the state:  let newValue = {value:'Buggy'};

But the option does not get selected.

Thank you!

**Comment by DanishOnGit** [comment]
Here is how I did it:
Below u can control the value of the `value` prop using some state value.
The thing is that I passed an options array and chose a value from that array!
`  

const dummy=[{label:'Grade A',value:'Grade 10-A'}]

<Select
            components={{ IndicatorSeparator: () => null }}
            value={dummy[0]}
            controlShouldRenderValue={true}
            placeholder='All Classrooms'
            className='classHome-select'
            classNamePrefix='classHome-select'
            styles={newStyles}
            options={dummy}
            onChange={(batch, action) => {
               
            }}          
            />`
